### DISCUSSION TITLE
Is Platinum tier (Siloed Tenant) really needed in Serverless?

### DISCUSSION THREAD

**Discussion by metarama** [aws-samples/aws-saas-factory-ref-solution-serverless-saas#48] [Q&A] [answered]
The 'noisy neighbor' and other issues that motivated the platinum or siloed tenant tier are more of a problem in the traditional (not serverless) architecture.

Given the 'infinite' scalability of Lambda, DynamoDB, and other serverless infrastructure I am wondering if I really need the 'Platinum' tier.

If the client insists physical separation of their data then Siloed DynamoDB tables makes sense.

**Comment by anubhavynr** [comment, marked as answer]
It is more than noisy neighbor - other factors that can influence is compliance, reducing blast radius, global reach (multi-region), tiering strategy that might require you to customize settings like provisioned concurrency/reserved concurrency etc. At times SaaS providers may leverage to store data in /tmp directory that can also get shared across invocations.

But you are right that most SaaS providers and their customers are concerned about data isolation instead of compute level isolation. In this case the common compute infra code can live inside bootstrap/shared template yaml file and tenant specific code remains inside tenant yaml file.
